web/src/app/modules/actions/modules/export-testspecification-button/components/export-testspecification-button.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { TestSpecification } from '../../../../../model/TestSpecification';
+import { TestCase } from '../../../../../model/TestCase';
+import { TestParameter } from '../../../../../model/TestParameter';
+import { ParameterAssignment } from '../../../../../model/ParameterAssignment';
+import { Url } from '../../../../../util/url';
+import { Type } from '../../../../../util/type';
+import { IContainer } from '../../../../../model/IContainer';
+import { SpecmateDataService } from '../../../../data/modules/data-service/services/specmate-data.service';
+import { ValidationService } from '../../../../forms/modules/validation/services/validation.service';
+import { TranslateService } from '@ngx-translate/core';
+import { AuthenticationService } from '../../../../views/main/authentication/modules/auth/services/authentication.service';
+import { AccessRights } from '../../../../../model/AccessRights';
+import { UserToken } from '../../../../views/main/authentication/base/user-token';
+
+@Component({
+    moduleId: module.id.toString(),
+    selector: 'export-testspecification-button',
+    templateUrl: 'export-testspecification-button.component.html',
+    styleUrls: ['export-testspecification-button.component.css']
+})
+
+export class ExportTestspecificationButton {
+
+    @Input()
+    public testSpecification: TestSpecification;
+
+    private contents: IContainer[];
+    private testCases: TestCase[];
+    private testParameters: TestParameter[];
+    private parameterAssignments: ParameterAssignment[];
+    private finalCsvString: string;
+
+    constructor(
+        private dataService: SpecmateDataService,
+        private validation: ValidationService,
+        private translate: TranslateService,
+        private auth: AuthenticationService) { }
+
+    // Export Function
+    public exportTestSpecification(): void {
```

Comment:
I'd make this method async to make it easier to handle all the asynchronous calls inside. Like this:

```suggestion
    public async exportTestSpecification(): Promise<void> {
```